doc/validate_interface_name_length.md

```diff
@@ -0,0 +1,178 @@
+| 0.1  | 17/05/23 | Eden Grisaro | Initial version    |
+
+
+
+# Motivation
+
+The kernel can't configure interfaces with names that are longer than `IFNAMSIZ` (`IFNAMSIZ` is the maximum buffer size needed to hold an interface name, including its terminating zero byte). Since we have this limitation we want to add validation to check the interface's name size.
+
+
+
+# About this Manual
+
+This document provides an overview of the implementation of adding validation of the interface's name size.
+
+
+# Design
+
+## Add validation to the following CLI command
+
+These are the interfaces available to be configured by CLI, hence, sanity on the interface name is required:
+* Vxlan
+* Vlan
+* Vrf
+* Loopback
+* Subinterface
+* Portchannel
+
+For each one, we will add (or make sure we have) validation for the length of the interface name.
+
+The following interfaces don't have validation of the interface name in their CLI command therefore we will add one:
+
+### Vxlan
+The user can configure Vxlan via CLI command, for example: "config vxlan map add neighbor_adv 10 1". We will add a limitation in the CLI command that the vxlan_name length will not be greater than `IFNAMSIZ`. The validation will be added to the sonic-utilities/config/vxlan.py file in the "add_vxlan" function.
+```python
+if len(vxlan_name) > IFNAMSIZ:
+        ctx.fail("'vxlan_name' is too long! The 'vxlan_name' length needs to be less than 16 characters")
+```
+
+### Subinterface
+The user can configure Subinterface via CLI command, for example: "sudo config subinterface add Ethernet0.100". We will add a limitation in the CLI command that the subinterface_name length will not be greater than `IFNAMSIZ`. The validation will be added to the sonic-utilities/config/main.py file in the "add_subinterface" function.
+```python
+if len(subinterface_name) > IFNAMSIZ:
+        ctx.fail("'subinterface_name' is too long! The 'subinterface_name' length needs to be less than 16 characters")
+```
+
+The following interface already has validation and we don't need to add additional check:
+### Vlan
+The user can configure vlan via CLI command, for example: "config vlan add 100". Valid vlan name is "Vlan"+{vid}. There is a limitation in the CLI command that the vid will not be greater than 4094. This limitation promises us the vlan name max length is 8 characters and therefore will not cross the 'IFNAMSIZ'. (You can find the length validation in the file sonic-utilities/config/vlan.py in the "add_vlan" function).
+
+### Vrf
+The user can configure vrf via CLI command, for example: "config vrf add Vrf100". Valid vrf name must begin with 'Vrf' or named 'mgmt'/'management' in case of ManagementVRF. There is a limitation in the CLI command that the vrf name length will be less than 16. This limitation promises us the vlan name length will not cross the `IFNAMSIZ`. (You can find the length validation in the file sonic-utilities/config/main.py in the "add_vrf" function).
```

Comment:
"promises us the vlan name length will not cross the `IFNAMSIZ`" which is 16 by definition